torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3252,6 +3252,28 @@ std::tuple<Tensor, Tensor> polar_backward(
   return std::make_tuple(grad_abs, grad_angle);
 }
 
+Tensor i1_backward(const Tensor& grad, const Tensor& self, const Tensor& result) {
+  return AT_DISPATCH_FLOATING_TYPES_AND2(
+      at::kHalf, at::kBFloat16, self.scalar_type(), "i1_backward", [&]() {
+        auto eps = std::numeric_limits<scalar_t>::epsilon();
+        auto self_is_not_tiny = self.abs() > eps;
+        auto safe_self = at::where(self_is_not_tiny, self, at::full({}, eps, self.options()));
```

Comment:
(Note to self): Following used to work. (investigate)
auto safe_self = at::where(self_is_not_tiny, self, eps);